docs/upgrade/3.1.rst

```diff
@@ -160,3 +168,54 @@ just an image as previously).
 Some user interfaces have some plugins hidden from display in edit/preview mode.
 :ttag:`render_plugin_block` provides a way to expose them for editing, and also more generally
 provides an alternative means of triggering a plugin's change form.
+
+
+Remove deprecations
+===================
+
+* ``cms.context_processor.media`` has been removed in favor of
+  ``cms.context_processor.cms_settings``, update your settings accordingly;
+
+* Old-style plugin table names are no longer supported. Relevant code has been removed; any
+  plugin table name must be migrated to the standard table names layout;
+
+
+.. _upgrade-guide-3.1:
+
+***********************************
+Upgrading for django CMS 3.0 to 3.1
+***********************************
+
+The two major changes regarding 3.1 are the removal of django-mptt in favor of django-treebeard
+and the dropped support for Django prior 1.6.
+
+Preliminary steps
+=================
+
+Before upgrading, please make sure that you current database is consistent.
+
+To ensure this, run two commands:
+
+* ``python manage.py cms delete_orphaned_plugins``
+* ``python manage.py fix_mptt``
+
+Settings update
+===============
+
+* Change ``cms.context_processor.media`` in ``cms.context_processor.cms_settings`` in
+  ``TEMPLATE_CONTEXT_PROCESSORS``;
+* Add ``treebeard`` in ``INSTALLED_APPS`` and remove ``mptt`` if not required by other
+  applications;
+* If using Django 1.7 remove ``cms`` and ``menus`` from ``MIGRATION_MODULES`` to support
+  the new migration layout;
+* If migrating for Django 1.6 and below to Django 1.7, remove ``south`` from ``installed_apps``;
+
+Update the database
+===================
+
+The migration for mptt to treebeard is handled by the migrations, so just running them will
+update your database:
+
+* Rename the plugin table names, either by applying relevant migrations or renaming the table
+  manually;
+* ``python manage.py migrate``;
```

Comment:
This is not clear. Should the user run migrations, rename tables manually, or both?